From d3e7ad68724a783f1a8ddf2f0692e16ee411d485 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Mon, 31 May 2010 16:42:20 -0400
Subject: [PATCH] revoke: export do_fsync

do_fsync is not typically exported, but revoke needs to use this
outside of sync.c.

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 fs/sync.c          |    3 ++-
 include/linux/fs.h |    1 +
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/fs/sync.c b/fs/sync.c
index 92b2281..c655e36 100644
--- a/fs/sync.c
+++ b/fs/sync.c
@@ -264,7 +264,7 @@ int vfs_fsync(struct file *file, struct dentry *dentry, int datasync)
 }
 EXPORT_SYMBOL(vfs_fsync);
 
-static int do_fsync(unsigned int fd, int datasync)
+int do_fsync(unsigned int fd, int datasync)
 {
 	struct file *file;
 	int ret = -EBADF;
@@ -276,6 +276,7 @@ static int do_fsync(unsigned int fd, int datasync)
 	}
 	return ret;
 }
+EXPORT_SYMBOL(do_fsync);
 
 SYSCALL_DEFINE1(fsync, unsigned int, fd)
 {
diff --git a/include/linux/fs.h b/include/linux/fs.h
index f7da001..2603135 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -2100,6 +2100,7 @@ extern int filemap_fdatawrite_range(struct address_space *mapping,
 extern int vfs_fsync_range(struct file *file, struct dentry *dentry,
 			   loff_t start, loff_t end, int datasync);
 extern int vfs_fsync(struct file *file, struct dentry *dentry, int datasync);
+extern int do_fsync(unsigned int fd, int datasync);
 extern int generic_write_sync(struct file *file, loff_t pos, loff_t count);
 extern void sync_supers(void);
 extern void emergency_sync(void);
-- 
1.6.5.2

